d2l-search-facets-grouping.html

```diff
@@ -0,0 +1,135 @@
+<link rel="import" href="../polymer/polymer-element.html">
+<link rel="import" href="../d2l-inputs/d2l-input-checkbox-spacer.html">
+<link rel="import" href="../d2l-typography/d2l-typography-shared-styles.html">
+
+<dom-module id="d2l-search-facets-grouping">
+	<template strip-whitespace>
+		<style>
+			:host {
+				display: flex;
+				flex-direction: column;
+			}
+			.d2l-search-facets-grouping-label {
+				@apply --d2l-body-compact-text;
+				font-weight: bold;
+			}
+
+			.d2l-search-facets-grouping-show-more > button {
+				@apply --d2l-body-compact-text;
+				border: none;
+				color: var(--d2l-color-celestine);
+				font-family: inherit;
+				outline: none;
+				padding: 0px;
+			}
+
+			.d2l-search-facets-grouping-show-more > button:hover,
+			.d2l-search-facets-grouping-show-more > button:focus {
+				cursor: pointer;
+				font-weight: bold;
+			}
+
+		</style>
+		<span class="d2l-search-facets-grouping-label">[[label]]</span>
+		<d2l-input-checkbox-spacer></d2l-input-checkbox-spacer>
+		<slot></slot>
+		<template is="dom-if" if="{{_showMoreButtonVisible}}">
+			<d2l-input-checkbox-spacer class="d2l-search-facets-grouping-show-more">
+				<button on-click="_showMore">More</button>
```

Comment:
I'm not sure what the design calls for in terms of what this button should look like... but I think we should be using one of our existing button styles instead of inventing something new. `<d2l-button-subtle>` seems like potentially a good choice? Maybe using the `<d2l-more-less>` "more" icon?

Reply:
More/less may not be totally appropriate since it's all about expanding/collapsing a bunch of content, whereas you just want to hide it initially and then expand it forever I think? But under the covers it uses a subtle button, which might work well for you.

Reply:
I can confirm - Hide initially, expand _forever_ is the desired behaviour for this control as part of the "Facetted Search Panel" pattern.

The "expand forever" behaviour may be desired by several other components as part of the MORE/LESS pattern, and might be worth doing consistently in the MORE/LESS control.

(Note the example of the GitHub "5 Invisible Conversations" & "Show" pattern on this page)

Reply:
Oh neat, yeah that could be a cool addition to more/less. The other potentially limiting factor for more/less is that today the only way to set a threshold is via pixel height -- e.g. "clip this at a height of 200px". So only showing 3 or 5 or 10 children isn't an option currently. But it could be added to more/less probably easily-ish.